### PR TITLE
Core/Movement: added safety checks

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
**Changes proposed:**

-  added some safety checks in order to prevent crashes

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [x] master

**Tests performed:** builds

### CODE REVIEW

File: src/server/game/Movement/MotionMaster.cpp [hunk 1]
```diff
@@ -58,9 +58,15 @@ void MotionMaster::InitDefault()
...
+
+        if (movement)
+        {
+            Mutate(movement, MOTION_SLOT_IDLE);
+            return;
+        }
     }
-    else
+
+    if (&si_idleMovement)
```
Comment (Shauren) [inline src/server/game/Movement/MotionMaster.cpp]: hahahahahaha. no.

Comment (ShinDarth) [inline src/server/game/Movement/MotionMaster.cpp]: ok... can you elaborate? ^^

Comment (Shauren) [inline src/server/game/Movement/MotionMaster.cpp]: You are taking the address of a variable, it will never be null.

**Comment by Shauren** [comment]
Does not solve the crash it is attempting to fix http://paste.ubuntu.com/23448340/

**Pull request closed without merging**